File: 0729_My Calendar I/MyCalendarI.java
class MyCalendar {
    private TreeMap<Integer, Integer> _booked;
    public MyCalendar() {
        _booked = new TreeMap<>();
    }

    public boolean book(int start, int end) {
        //The floorKey() method is used to return the greatest key less than or equal to given key from the parameter.
        Integer lb = _booked.floorKey(start);
        //get end mapped by lb
        if(lb != null && _booked.get(lb) > start) return false;
        //ceilingKey() returns the least key greater than or equal to the given key or null if the such a key is absent.
        Integer ub = _booked.ceilingKey(start);
        if(ub != null && ub < end) return false;
        _booked.put(start, end);
        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * boolean param_1 = obj.book(start,end);
 */

File: 0729_My Calendar I/MyCalendarI.cs
public class MyCalendar {

    private List<Tuple<int,int>> list = new List<Tuple<int,int>>();
    public MyCalendar() {
        
    }
    
    public bool Book(int start, int end) {
        foreach(var t in list){
            if(Math.Max(start, t.Item1) < Math.Min(end, t.Item2)) return false;
        }

        list.Add(new Tuple<int,int>(start,end));
        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * bool param_1 = obj.Book(start,end);
 */